### PR TITLE
MDEV-17458 Unable to start galera node

### PR DESCRIPTION
**Pull request by sciascid**
Bootstrapping a new cluster from a backup created from a MariaDB
version prior to 10.3.5, may result in error "SST position can't be
set in past" when attempting to join additional nodes.
The problem stems from the fact that when reading the wsrep position
from InnoDB, the position is looked up in two places:
the system header, where versions prior to 10.3.5 used to store
WSREP's position; and rollback segments, this is where newer versions
store the position.
When starting a new cluster, the starting seqno is 0 and a new cluster
UUID is generated. This is persisted in rollback segments, but the old
UUID and seqno are not cleared from the sys header.
Subsequently, when reading back the position,
`trx_rseg_read_wsrep_checkpoint` is going to return the maximum seqno
found in both system header and rollback segments. So in the case of a
newly bootstrapped cluster, it's always going to return the old
cluster information.
The fix consists of changing `trx_rseg_read_wsrep_checkpoint` so that
only rollback segments are looked up. On startup, the sys header
position is read, and if present, it is copied to rollback segments
unless a newer position is already present in the rollback segments.
Finally the sys header position is cleared.

### CODE REVIEW

**Comment by sciascid** [comment]
galera suites: https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/692/consoleFull
default suites: https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/691/console

**Comment by sciascid** [comment]
galera_3nodes.galera_safe_to_bootstrap passes, after backporting fixes from 10.4.
new mtr run: https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/696/consoleFull
and for comparison, mtr run with mariadb 10.3: https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/698/

File: storage/innobase/trx/trx0rseg.cc [hunk 1]
```diff
@@ -107,18 +127,15 @@ void trx_rseg_update_wsrep_checkpoint(const XID* xid)
...
+	if (must_clear_rsegs) {
 		/* Because the UUID part of the WSREP XID differed
 		from current_xid_uuid, the WSREP group UUID was
 		changed, and we must reset the XID in all rollback
 		segment headers. */
 		for (ulint rseg_id = 1; rseg_id < TRX_SYS_N_RSEGS; ++rseg_id) {
 			if (const trx_rseg_t* rseg =
 			    trx_sys.rseg_array[rseg_id]) {
-				trx_rseg_update_wsrep_checkpoint(
+				trx_rseg_write_wsrep_checkpoint(
```
Comment (temeo) [inline storage/innobase/trx/trx0rseg.cc]: Should we call here `trx_rseg_clear_wsrep_checkpoint()` similar to 10.4 and implement `trx_rseg_clear_wsrep_checkpoint()` by writing null XID?  `mlog_memset()` which is used in 10.4 is not available in 10.3 to my knowledge.

Comment (sciascid) [reply storage/innobase/trx/trx0rseg.cc]: OK, done. Test running here: https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/708/

Comment (sciascid) [reply storage/innobase/trx/trx0rseg.cc]: Made one small change to trx_rseg_clear_wsrep_checkpoint(). Running tests again here https://jenkins.galeracluster.com:8443/view/mtr/job/mtr-galera-4.x-mariadb-10.4/709/

**Pull request closed without merging**